core/src/test/java/io/aiven/kafka/tieredstorage/chunkmanager/cache/DiskBasedChunkCacheMetricsTest.java

```diff
@@ -0,0 +1,116 @@
+import io.aiven.kafka.tieredstorage.manifest.SegmentManifest;
+
+import org.assertj.core.api.InstanceOfAssertFactories;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class DiskBasedChunkCacheMetricsTest {
+    static final MBeanServer MBEAN_SERVER = ManagementFactory.getPlatformMBeanServer();
+
+    public static final Uuid SEGMENT_ID = Uuid.randomUuid();
+
+    static final int LOG_SEGMENT_BYTES = 10;
+    static final RemoteLogSegmentMetadata REMOTE_LOG_SEGMENT_METADATA =
+        new RemoteLogSegmentMetadata(
+            new RemoteLogSegmentId(
+                new TopicIdPartition(Uuid.randomUuid(), new TopicPartition("topic", 0)),
+                SEGMENT_ID),
+            1, -1, -1, -1, 1L,
+            LOG_SEGMENT_BYTES, Collections.singletonMap(1, 100L));
+
+    @Mock
+    ChunkManager chunkManager;
+    @Mock
+    SegmentManifest segmentManifest;
+    @TempDir
+    Path baseCachePath;
+
+    DiskBasedChunkCache diskBasedChunkCache;
+
+    @BeforeEach
+    void setUp() {
+        diskBasedChunkCache = new DiskBasedChunkCache(chunkManager);
+        diskBasedChunkCache.configure(Map.of(
+            "retention.ms", "-1",
+            "size", "-1",
+            "path", baseCachePath.toString()
+        ));
+    }
+
+    @Test
+    void shouldRecordMetrics() throws Exception {
+        when(chunkManager.getChunk(any(), eq(segmentManifest), eq(0)))
+            .thenReturn(new ByteArrayInputStream("test".getBytes()));
+
+        diskBasedChunkCache.getChunk(REMOTE_LOG_SEGMENT_METADATA, segmentManifest, 0);
+        diskBasedChunkCache.getChunk(REMOTE_LOG_SEGMENT_METADATA, segmentManifest, 0);
+
+        final var objectName = new ObjectName("aiven.kafka.server.tieredstorage.cache:type=chunk-cache");
+        assertThat(MBEAN_SERVER.getAttribute(objectName, "cache-hits-total"))
+            .isEqualTo(1.0);
+        assertThat(MBEAN_SERVER.getAttribute(objectName, "cache-misses-total"))
+            .isEqualTo(1.0);
+        assertThat(MBEAN_SERVER.getAttribute(objectName, "cache-load-success-time-total"))
+            .asInstanceOf(InstanceOfAssertFactories.DOUBLE)
+            .isGreaterThan(0);
+
+        // compute is considered as load success regardless if present or not
+        assertThat(MBEAN_SERVER.getAttribute(objectName, "cache-load-success-total"))
+            .isEqualTo(2.0);
+        assertThat(MBEAN_SERVER.getAttribute(objectName, "cache-load-failure-time-total"))
+            .asInstanceOf(InstanceOfAssertFactories.DOUBLE)
+            .isEqualTo(0);
```

Comment:
Nit: should we preserve the same style and if `asInstanceOf` is not needed just do not add it? `isEqualTo(0.0)` would work just fine here as below. Or otherwise using `asInstanceOf` everywhere? WDYT?